project-management/Github.md

```diff
@@ -0,0 +1,77 @@
+# Managing public (or private) projects using GitHub
+
+We use a common set of labels across projects in GitHub. It is
+possible to precisely control this set of labels using
+the [settings GitHub app][github-app-settings] (needs to be enabled
+for your org). Use the [`settings.yml`](../.github/settings.yml) from
+this repository as a starting point.
+
+[github-app-settings]: https://github.com/probot/settings
+
+## Labels
+
+We use labels to classify tasks by priority, from P0 to P4:
+
+* **P0 (blocker)** Do not do anything else, this is a blocker.
+* **P1 (critical)** Must have.
+* **P2 (major)** Should have.
+* **P3 (minor)** Could have.
+* **P4 (trivial)** Someday/maybe. Consider closing as "wontfix".
+
+Labels are also useful to classify issues by type or by component. Use
+colours to distinguish whether a label designates a type, a component
+or something else.
+
+## Project leads
+
+Each project must have a project lead (maintainer). By convention, the
+PL is whoever appears at the top of [`CODEOWNERS`][github-codeowners]
+file, e.g.:
+
+```
+* @mboes
+```
+
+This has the effect that @mboes will automatically be tagged as
+a reviewer for all incoming PR's, unless later lines in the same file
+override that.
+
+[github-codeowners]: https://help.github.com/articles/about-codeowners/
+
+## Milestones
+
+We gate tickets with milestones. Milestones always have an associated
+due date.
+
+## Releases
+
+Every public release must include a `CHANGELOG.md` file, in the format
+of [keepachangelog]. New entries to the changelog are integral parts
+of bug fix and feature PR's.
+
+Each release has an associated tag of the form `v$VERSION`, GPG-signed
+by the project lead (use `git tag -s`).
```

Comment:
This is new, isn't it?

I've never really cared much for pgp, as you probably know. But I won't fight this either.

On the other hand, I think that if we are to mandate this, we should have a gpg guide as well, with easy step to do just what we need here:

- Generate a key pair with gpg
- Importing a key pair with gpg on another computer
- Uploading a key on github
- Is the something I'm forgetting

(keys don't need to be uploaded to a key sever for this purpose, so we probably can keep this out)

Reply:
GPG signing has been supported by Git since at least a few years. GPG signing is common practice and should be part of every engineers toolbelt by now. As for everything else on GitHub, there exists help pages for every topic you mention for engineers that really are lost, and plenty of GPG specific guides that I think we should not duplicate. See e.g. https://help.github.com/articles/signing-commits-using-gpg/. I'll add a link to the top-level help in case engineers forget that this resource exists.

Reply:
By new I mean: we haven't been doing this.

> GPG signing is common practice and should be part of every engineers toolbelt by now.

Yet, I sincerely believe this is not even approximately the case. The Github help page will help (and it's arguably necessary to link to them), but anything which can make the process more standardised is welcome: it takes problem off the minds of engineers, and it helps ensure that the process that engineer follow actually meet the company expectations.

Reply:
Writing yet another GPG guide is not a good use of my time, and I doubt that it's a good use of yours. I'll add a link to one of the many guides out there for informational purposes, e.g. this one: https://github.com/pstadler/keybase-gpg-github.

> anything which can make the process more standardised is welcome

It doesn't matter how you publish your identity (which in fact is independent from the company). What does matter is that exists.

Reply:
Sure. This one does look ok, though it relies on this keybase tool that I didn't know about. We can ask around for a better guide.

Reply:
In the very first section of that guide, it points to [another guide](https://help.github.com/articles/generating-a-new-gpg-key/) that doesn't mention keybase, for the heretics that would prefer to continue living in the pre-keybase era. That other guide is.... a GitHub help page!